support-frontend/app/controllers/PayPalRegular.scala

```diff
@@ -54,6 +54,16 @@ class PayPalRegular(
     }
   }
 
+  def createAgreementAndRetrieveUser: Action[Token] = maybeAuthenticatedAction().async(circe.json[Token]) { implicit request =>
+    withPaypalServiceForRequest(request) { service =>
+      service.createAgreementAndRetrieveUser(request.body)
+    }.map { maybePayPalCheckoutDetails =>
+      maybePayPalCheckoutDetails
+        .map(details => Ok(details.asJson))
+        .getOrElse(BadRequest("We were unable to create an agreement for this request (missing user details or baid)"))
```

Comment:
if this endpoint fails do we need to alarm?

Reply:
We never have done previously. Currently the user will see this error message:
<img width="402" alt="Screen Shot 2021-09-27 at 17 45 13" src="https://user-images.githubusercontent.com/181371/134951142-b842be5b-31a9-4990-ac9e-e706463c5197.png">
I think that's sufficient?

Reply:
for the AB test it probably is okay, but we should return a specific error message if there is a known case where the user needs to take action. And for everything else, we should be aware of the issue and look into it.  This comment is mainly in the light of the bugs we found this last week where people are seeing this error message due to gaps in the logic.